code/modules/mob/living/simple_animal/hostile/headcrab.dm

```diff
@@ -191,3 +198,390 @@
+		to_chat(C, "<span class='userdanger'>You have been splashed with acid!</span>")
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/mode_switch(mode_to_switch, sound_to_play, clear_targets = TRUE, change_icon)
+	debug_gonarch_modeswitches.Add(mode_to_switch) //TODO: Remove simple debug
+	if(mode_to_switch == mode)
+		return
+	mode = mode_to_switch
+	if(clear_targets)
+		LoseTarget()
+	if(sound_to_play)
+		var/sound_picked = pick(sound_to_play)
+		playsound(src, sound_picked, 200, TRUE, 15, pressure_affected = FALSE)
+	if(change_icon)
+		icon_state = change_icon
+		icon_living = change_icon
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/gonarch_rest()
+	if(get_area(src) == home_nest_area) // If I am home & Everything is calm
+		if(aggro_vision_range != gonarch_aggro_vision_range || vision_range != gonarch_standard_vision_range || speed != gonarch_standard_speed)
+			aggro_vision_range = gonarch_aggro_vision_range
+			vision_range = gonarch_standard_vision_range
+			speed = gonarch_standard_speed
+		if(homesick_level)
+			homesick_level = 0
+			step_towards(src, home_nest_turf, speed)
+		try_make_headcrabs() //Its its own proc, so they can be called by admins.
+	else
+		if(homesick_level >= 5)
+			mode_switch(GONARCH_MODE_RETURNING, gonarch_soundlist_return, change_icon = "gonarch_peaceful")
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/gonarch_return()
+	speed = -1
+	var/turf/olddist = get_dist(src, home_nest_turf)
+	step_towards(src, home_nest_turf, speed)
+	target = home_nest_turf
+	if(get_dist(src, target) >= olddist)
+		frustration++
+		if(frustration > frustration_limit)
+			DestroyPathToTarget()
+			homesick_level++
+	else
+		frustration = 0
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/gonarch_newhome()
+	var/list/turfs = list()
+	var/list/turfs_in_range = range(4, src)
+	for(var/turf/T in range(gonarch_nestfind_range, src))
+		if(T in turfs_in_range)
+			continue
+		if(isspaceturf(T))
+			continue
+		if(T.density)
+			continue
+		var/lightingcount = T.get_lumcount(0.5) * 10
+		if(lightingcount > 1 && gonarch_finding_home_attempts < gonarch_finding_home_limit_attempts)
+			continue
+		turfs += T
+	if(!length(turfs))
+		gonarch_nestfind_range += 5
+		gonarch_finding_home_attempts++
+		return
+	gonarch_finding_home_attempts = 0
+	home_nest_turf = pick(turfs)
+	home_nest_area = get_area(home_nest_turf)
+	mode_switch(GONARCH_MODE_RETURNING, gonarch_soundlist_return, change_icon = "gonarch_peaceful")
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/gonarch_rampage()
+	//We want to lose all targets, and get the hell out of here. Vision/care is restored when back in base.
+	aggro_vision_range = 0
+	vision_range = 0
+	LoseTarget()
+	homesick_level = 0
+	for(var/mob/living/carbon/C in hearers(gonarch_screech_range, src))
+		to_chat(C, "<span class='warning'><font size='3'><b>You hear a ear piercing shriek and your senses dull!</font></b></span>")
+		C.Weaken(25)
```

Comment:
```suggestion
		C.Weaken(6)
```
This needs to be VASTLY reduced, you get caught by this and it's just straight up game over. No chance to counterplay, just death.

Reply:
is there any reason for it to weaken *and* stun? I am pretty sure weaken does everything a stun would do so having both values is redundant.